### PR TITLE
Added Bonestorm, Fixed Desert Wolf + other minor changed. Closes #110 & #143 & #147

### LINKED ISSUE #110: Bonestorms should be added.

### PR DESCRIPTION
**Pull request by GirafiStudios**
#145 have kinda been handled too, at least improvements. Not fixed tho.

### CODE REVIEW

**Comment by Shadowclaimer** [comment]
Commenting to note I worked with Girafi step by step for all these changes and any off the wall weird ones that don't seem related were me telling him to fix something while he was in that file.

=)

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 1]
```diff
@@ -23,24 +24,15 @@
 public class EntityDesertWolf extends EntityTameable {
     private float field_70926_e;
     private float field_70924_f;
-
-    /**
-     * true is the wolf is wet else false
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 2]
```diff
@@ -23,24 +24,15 @@
...
     private boolean isShaking;
     private boolean field_70928_h;
-
-    /**
-     * This time increases while wolf is shaking and emitting water particles.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 3]
```diff
@@ -63,33 +55,27 @@ public EntityDesertWolf(World par1World) {
...
+        if (this.isTamed()) {
+            this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(20.0D);
+        } else {
+            this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(8.0D);
+        }
     }
 
-    /**
-     * Returns true if the newer Entity AI code should be run
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 4]
```diff
@@ -63,33 +55,27 @@ public EntityDesertWolf(World par1World) {
...
     @Override
     public boolean isAIEnabled() {
         return true;
     }
 
-    /**
-     * Finds the closest player within 16 blocks to attack, or null if this
-     * Entity isn't interested in attacking (Animals, Spiders at day, peaceful
-     * PigZombies).
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 5]
```diff
@@ -63,33 +55,27 @@ public EntityDesertWolf(World par1World) {
...
     @Override
     protected Entity findPlayerToAttack() {
         EntityPlayer entityplayer = this.worldObj.getClosestVulnerablePlayerToEntity(this, 16.0D);
         return entityplayer != null && this.canEntityBeSeen(entityplayer) ? entityplayer : null;
     }
 
-    /**
-     * Sets the active target the Task system uses for tracking
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 6]
```diff
@@ -99,18 +85,20 @@ public void setAttackTarget(EntityLivingBase par1EntityLiving) {
         }
     }
 
-    /**
-     * main AI tick function, replaces updateEntityActionState
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 7]
```diff
@@ -99,18 +85,20 @@ public void setAttackTarget(EntityLivingBase par1EntityLiving) {
...
     @Override
-    protected void updateAITick() {
+    protected void updateAITick()
+    {
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert unnecessary change.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 8]
```diff
@@ -99,18 +85,20 @@ public void setAttackTarget(EntityLivingBase par1EntityLiving) {
...
         this.dataWatcher.updateObject(18, Float.valueOf(this.getHealth()));
     }
 
-    /**
-     * Checks if the entity's current position is a valid location to spawn this
-     * entity.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 9]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
         this.playSound("mob.wolf.step", 0.15F, 1.0F);
     }
 
-    /**
-     * (abstract) Protected helper method to write subclass entity data to NBT.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 10]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
-        par1NBTTagCompound.setBoolean("Angry", this.isAngry());
+    public void writeEntityToNBT(NBTTagCompound nbtTagCompound) {
+        super.writeEntityToNBT(nbtTagCompound);
+        nbtTagCompound.setBoolean("Angry", this.isAngry());
+        nbtTagCompound.setByte("CollarColor", (byte) this.getCollarColor());
     }
 
-    /**
-     * (abstract) Protected helper method to read subclass entity data from NBT.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 11]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
+
+        if (nbtTagCompound.hasKey("CollarColor", 99)) {
+            this.setCollarColor(nbtTagCompound.getByte("CollarColor"));
+        }
 
     }
 
-    /**
-     * Determines if an entity can be despawned, used on idle far away entities
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 12]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
     @Override
     protected boolean canDespawn() {
-        return this.isAngry();
+        return this.isAngry() && this.ticksExisted > 2400;
     }
 
-    /**
-     * Returns the sound this mob makes while it's alive.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 13]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
     @Override
     protected String getLivingSound() {
-        return this.isAngry() ? "mob.wolf.growl" : (this.rand.nextInt(3) == 0 ? (this.isTamed() && this.dataWatcher.getWatchableObjectInt(18) < 10 ? "mob.wolf.whine" : "mob.wolf.panting") : "mob.wolf.bark");
+        return this.isAngry() ? "mob.wolf.growl" : (this.rand.nextInt(3) == 0 ? (this.isTamed() && this.dataWatcher.getWatchableObjectFloat(18) < 10.0F ? "mob.wolf.whine" : "mob.wolf.panting") : "mob.wolf.bark");
     }
 
-    /**
-     * Returns the sound this mob makes when it is hurt.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 14]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
     @Override
     protected String getHurtSound() {
         return "mob.wolf.hurt";
     }
 
-    /**
-     * Returns the sound this mob makes on death.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 15]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
     @Override
     protected String getDeathSound() {
         return "mob.wolf.death";
     }
 
-    /**
-     * Returns the volume for the sounds this mob makes.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 16]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
     @Override
     protected float getSoundVolume() {
         return 0.4F;
     }
 
-    /**
-     * Drop 0-2 items of this living's type. @param par1 - Whether this entity
-     * has recently been hit by a player. @param par2 - Level of Looting used to
-     * kill this mob.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 17]
```diff
@@ -121,86 +109,57 @@ protected void func_145780_a(int p_145780_1_, int p_145780_2_, int p_145780_3_,
...
             int amount = rand.nextInt(2) + 1;
             this.dropItem(AtumItems.ITEM_PELT, amount);
         }
     }
 
-    /**
-     * Called frequently so the entity can update its state every tick as
-     * required. For example, zombies and skeletons use this to react to
-     * sunlight and start to burn.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 18]
```diff
@@ -213,9 +172,6 @@ public void onLivingUpdate() {
         }
     }
 
-    /**
-     * Called to update the entity's position/logic.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 19]
```diff
@@ -276,9 +232,6 @@ public boolean getWolfShaking() {
     }
 
     @SideOnly(Side.CLIENT)
-    /**
-     * Used when calculating the amount of shading to apply while the wolf is shaking.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 20]
```diff
@@ -306,18 +259,11 @@ public float getEyeHeight() {
         return this.height * 0.8F;
     }
 
-    /**
-     * The speed it takes to move the entityliving's rotationPitch through the
-     * faceEntity method. This is only currently use in wolves.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 21]
```diff
@@ -306,18 +259,11 @@ public float getEyeHeight() {
...
     @Override
     public int getVerticalFaceSpeed() {
         return this.isSitting() ? 20 : super.getVerticalFaceSpeed();
     }
 
-    /**
-     * Called when the entity is attacked.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 22]
```diff
@@ -335,33 +281,52 @@ public boolean attackEntityFrom(DamageSource par1DamageSource, float par2) {
...
+            this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(20.0D);
+        } else {
+            this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(8.0D);
+        }
     }
 
-    /**
-     * Called when a player interacts with a mob. e.g. gets milk from a cow,
-     * gets into the saddle on a pig.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 23]
```diff
@@ -418,36 +382,23 @@ public void handleHealthUpdate(byte par1) {
...
     @SideOnly(Side.CLIENT)
     public float getTailRotation() {
-        return this.isAngry() ? 1.5393804F : (this.isTamed() ? (0.55F - (float) (20 - this.dataWatcher.getWatchableObjectInt(18)) * 0.02F) * (float) Math.PI : ((float) Math.PI / 5F));
+        return this.isAngry() ? 1.5393804F : (this.isTamed() ? (0.55F - (20.0F - this.dataWatcher.getWatchableObjectFloat(18)) * 0.02F) * (float) Math.PI : ((float) Math.PI / 5F));
     }
 
-    /**
-     * Checks if the parameter is an item which this animal can be fed to breed
-     * it (wheat, carrots or seeds depending on the animal type)
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 24]
```diff
@@ -418,36 +382,23 @@ public void handleHealthUpdate(byte par1) {
...
     @Override
     public boolean isBreedingItem(ItemStack par1ItemStack) {
         return par1ItemStack == null ? false : (!(par1ItemStack.getItem() instanceof ItemFood) ? false : ((ItemFood) par1ItemStack.getItem()).isWolfsFavoriteMeat());
     }
 
-    /**
-     * Will return how many at most can spawn in a chunk at once.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 25]
```diff
@@ -418,36 +382,23 @@ public void handleHealthUpdate(byte par1) {
...
     @Override
     public int getMaxSpawnedInChunk() {
         return 8;
     }
 
-    /**
-     * Determines whether this wolf is angry or not.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 26]
```diff
@@ -418,36 +382,23 @@ public void handleHealthUpdate(byte par1) {
...
     public boolean isAngry() {
         return (this.dataWatcher.getWatchableObjectByte(16) & 2) != 0;
     }
 
-    /**
-     * Sets whether this wolf is angry or not.
-     */
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert comment removal.

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 27]
```diff
@@ -480,30 +418,33 @@ public boolean canMateWith(EntityAnimal par1EntityAnimal) {
...
 
-        return entitywolf;
+        return entityDesertWolf;
     }
 
-    @Override
-    public EntityLivingBase getOwner() {
-        return null;
+    public boolean func_70922_bv() {
+        return this.dataWatcher.getWatchableObjectByte(19) == 1;
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert moving of the function,
Old location: https://github.com/TeamMetallurgy/Atum/pull/148/files#diff-3a33804d239ffb942f87589ddf58ab6dL488

File: src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java [hunk 28]
```diff
@@ -480,30 +418,33 @@ public boolean canMateWith(EntityAnimal par1EntityAnimal) {
...
     }
-}
+}
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/entity/EntityDesertWolf.java]: Kindly revert unnecessary change.

File: src/main/java/com/teammetallurgy/atum/handler/CraftingHandler.java [hunk 29]
```diff
@@ -97,4 +98,5 @@ private void addShapelessRecipes() {
         GameRegistry.addShapelessRecipe(new ItemStack(Items.string, 3), AtumItems.ITEM_FLAX);
     }
 
-}
+
+}
```
Comment (Glassmaker) [inline src/main/java/com/teammetallurgy/atum/handler/CraftingHandler.java]: kindly revert necessary change

**Comment by Glassmaker** [comment]
Thanks for the fixes :)

**Pull request merged**